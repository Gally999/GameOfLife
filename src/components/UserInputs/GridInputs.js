import React from 'react';
import PropTypes from 'prop-types';

const GridInputs = ({ x, y, onDimensionsChange }) => {
  const genericSync = (event) => {
    onDimensionsChange({
      x: event.target.name === 'x' ? parseInt(event.target.value, 10) : x,
      y: event.target.name === 'y' ? parseInt(event.target.value, 10) : y,
    });
  };

  return (
    <div>
      <label htmlFor="y">Height:
        <input
          type="number"
          value={y}
          id="y"
          name="y"
          onChange={genericSync}
          required
        />
      </label>
      <div />
      <label htmlFor="x">Width:
        <input
          type="number"
          value={x}
          id="x"
          name="x"
          onChange={genericSync}
          required
        />
      </label>
    </div>
  );
};

GridInputs.propTypes = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  onDimensionsChange: PropTypes.func.isRequired,
};

export default GridInputs;
