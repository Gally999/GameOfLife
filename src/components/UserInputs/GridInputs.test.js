import React from 'react';
import GridInputs from './GridInputs';

describe('GridInputs', () => {
  it('renders the grid inputs component correctly', () => {
    const subject = (
      <GridInputs
        x={4}
        y={7}
        onDimensionsChange={() => {}}
      />
    );
    expect(subject).toMatchSnapshot();
  });
});
