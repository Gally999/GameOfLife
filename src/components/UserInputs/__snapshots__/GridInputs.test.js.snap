// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridInputs renders the grid inputs component correctly 1`] = `
<GridInputs
  onDimensionsChange={[Function]}
  x={4}
  y={7}
/>
`;
